pydicer/convert/pet_to_nifti_convert_new.py

```diff
@@ -0,0 +1,409 @@
+                                          RTSTRUCT. Defaults to ['GTVt'].
+        patient_weight_from_ct (float, optional): If the patient's weight is
+                                                  missing from the PT DICOM
+                                                  it can be provided through
+                                                  this argument. Defaults to
+                                                  None.
+        extension (str, optional): The extension in which to save the NIFTI.
+                                   Defaults to '.nii'.
+        dtype_image (numpy.dtype, optional): The dtype in which to save the
+                                             image. Defaults to np.float32.
+        dtype_mask (numpy.dtype, optional): The dtype in which to save the
+                                            segmentation. Defaults to np.uint8.
+    Raises:
+        MissingWeightException: Error to alert when the weight is missing from
+                                the PT, to compute the SUV.
+        RuntimeError: Error to alert when one or more slices are missing
+        ValueError: Raised when a modality or a unit (for the PT) is not
+                    handled.
+    Returns:
+        numpy.array: The numpy image, used to compute the bounding boxes
+    """
+    slices = [pdcm.read_file(dcm) for dcm in input_filepaths]
+    slices.sort(key=lambda x: float(x.ImagePositionPatient[2]))
+    if modality == 'PT':
+        if hasattr(slices[0], 'PatientWeight') and (slices[0].PatientWeight is None):
+            if hasattr(slices[0], 'PatientsWeight'):
+                patient_weight = float(slices[0].PatientsWeight)
+            elif patient_weight_from_ct is not None:
+                patient_weight = patient_weight_from_ct
+            else:
+                # raise MissingWeightException(
+                #     'Cannot compute SUV the weight is missing')  # from Shuchao's code
+                patient_weight = 75.0  # From Shuchao's code
+                warnings.warn(
+                    "Cannot find the weight of the patient, hence it "
+                    "is approximated to be 75.0 kg")
+        elif not hasattr(slices[0], 'PatientWeight'):
+            if hasattr(slices[0], 'PatientsWeight'):
+                patient_weight = float(slices[0].PatientsWeight)
+            elif patient_weight_from_ct is not None:
+                patient_weight = patient_weight_from_ct
+            else:
+                # raise MissingWeightException(
+                #     'Cannot compute SUV the weight is missing')  # from Shuchao's code
+                patient_weight = 75.0  # From Shuchao's code
+                warnings.warn(
+                    "Cannot find the weight of the patient, hence it "
+                    "is approximated to be 75.0 kg")
+        else:
+            patient_weight = float(slices[0].PatientWeight)
+
+    # Check if all the slices come from the same serie

```

Comment:
This kind of check should be done in the preprocess module. We sort out DICOM files based on the SeriesInstanceUID